provider/router/routes.go

```diff
@@ -305,39 +292,43 @@ func UninstallApp(c *gin.Context) {
 		if slices.Contains(installedApps, payloadJson.App) {
 			err = devices.UninstallApp(dev, payloadJson.App)
 			if err != nil {
-				c.JSON(http.StatusInternalServerError, gin.H{"error": fmt.Sprintf("Failed to uninstall app `%s`", payloadJson.App)})
+				api.GenericResponse(c, http.StatusInternalServerError, fmt.Sprintf("Failed to uninstall app `%s`", payloadJson.App), installedApps)
 				return
 			}
-			c.JSON(http.StatusOK, gin.H{"message": fmt.Sprintf("Successfully uninstalled app `%s`", payloadJson.App)})
+			deletedAppIndex := slices.Index(installedApps, payloadJson.App)
+			if deletedAppIndex != -1 {
+				installedApps = append(installedApps[:deletedAppIndex], installedApps[deletedAppIndex+1:]...)
+			}
+			api.GenericResponse(c, http.StatusOK, fmt.Sprintf("Successfully uninstalled app `%s`", payloadJson.App), installedApps)
 			return
 		}
-		c.JSON(http.StatusBadRequest, gin.H{"error": fmt.Sprintf("App `%s` is not installed on device", payloadJson.App)})
+		api.GenericResponse(c, http.StatusBadRequest, fmt.Sprintf("App `%s` is not installed on device", payloadJson.App), installedApps)
 		return
 	}
 
-	c.JSON(http.StatusBadRequest, gin.H{"error": fmt.Sprintf("Device with udid `%s` does not exist", udid)})
+	api.GenericResponse(c, http.StatusBadRequest, fmt.Sprintf("Did not find device with udid `%s`", udid), nil)
 }
 
 func ResetDevice(c *gin.Context) {
 	udid := c.Param("udid")
 
 	if device, ok := devices.DBDeviceMap[udid]; ok {
 		if device.IsResetting {
-			c.JSON(http.StatusInternalServerError, gin.H{"error": "Device setup is already being reset"})
+			api.GenericResponse(c, http.StatusInternalServerError, "Device setup is already being reset", nil)
```

Comment:
It might be better to return a 409 Conflict error.

Reply:
Done [here](https://github.com/shamanec/GADS/pull/185/commits/4957832b1908943c42adb380c3ebcb9b9852e2b1)